Sources/TodoResource.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  TodoResource.swift
+//  TODOBackend
+//
+//  Created by Dan Appel on 5/1/16.
+//
+//
+
+import Resource
+import JSONMediaType
+
+func makeTodoResource(store todoStore: TodoStore) -> Resource {
+
+    return Resource(mediaTypes: [JSONMediaType()]) { todo in
+
+        // GET /
+        todo.index { request in
+            let todos = todoStore.getAll()
+            return Response(content: todos.content)
+        }
+
+        // POST /
+        todo.create(content: Todo.self) { request, todo in
+            let id = todoStore.nextId()
+            let newTodo = Todo(
+                id: id,
+                url: "http://01aa3281.ngrok.io/\(id)",
+                title: todo.title, completed: todo.completed, order: todo.order)
```

Comment:
Yeah this is the part I would do differently. You're treating the data you're receiving to construct a `Todo` as a `Todo` itself. But this is not a `Todo` what you're receiving is the information needed to construct a `Todo`. I would make `Todo` a `Struct` with no `Optional`s at all. I would create another `Struct`, something like `TodoInfo` (better name needed). And this `TodoInfo` would also not have any `Optional`.  I would also abstract the `id` generation inside `todoStore.insert(todo: newTodo)`. The `todoStore` would be responsible to receive `todoInfo` and return `todo` with the correct `id`.